src/Illuminate/Database/Eloquent/Concerns/HasUlidPrimaryKey.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Concerns;
+
+use Illuminate\Support\Str;
+
+trait HasUlidPrimaryKey
+{
+    /**
+     * Generate a primary ULID for the model.
+     *
+     * @return void
+     */
+    public static function bootHasUlidPrimaryKey()
+    {
+        static::creating(function (self $model) {
+            if (empty($model->{$model->getKeyName()})) {
+                $model->{$model->getKeyName()} = $model->generatePrimaryKey();
+            }
+        });
+    }
+
+    /**
+     * Generate the primary ULID key for the model.
+     *
+     * @return string
+     */
+    public function generatePrimaryKey()
+    {
+        return (string) Str::ulid();
```

Comment:
I didn't use `Str::orderedUlid()` as I'm not even sure that's possible with ULID?

Reply:
[ULID : Universally Unique Lexicographically Sortable Identifiers.](https://github.com/ulid/spec)

The abbreviation itself already suggests that it is ordered / sortable. 👍